src/test/java/br/com/caelum/clines/api/locations/LocationServiceTest.java

```diff
@@ -52,4 +57,39 @@ public void shouldThrowExceptionIfLocationDoesNotExists() {
                 () -> service.showLocationBy(1L)
         );
     }
+
+    @Test
+    void shouldListAllLocations() {
+        Location location1 = new Location(Country.BR, "SAO PAULO", "SAO PAULO");
+        Location location2 = new Location(Country.BR, "SAO PAULO", "CAMPINAS");
+        List<Location> locations = List.of(location1, location2);
+
+        given(repository.findAll()).willReturn(locations);
+
+        List<LocationView> locationViews = service.listAllLocations();
+
+        then(repository).should(only()).findAll();
+        then(mapper).should(times(2)).map(any(Location.class));
+
+        assertEquals(2, locationViews.size());
+
+        for (int i = 0; i < locationViews.size(); i++) {
+            assertEquals(locations.get(i).getId(), locationViews.get(i).getId());
+            assertEquals(locations.get(i).getCountry().getDescription(), locationViews.get(i).getCountry());
+            assertEquals(locations.get(i).getState(), locationViews.get(i).getState());
+            assertEquals(locations.get(i).getCity(), locationViews.get(i).getCity());
+        }
+    }
+
+    @Test
+    void shouldAnEmptyList() {
```

Comment:
This test name is a little bit weird, shouldn't be "shouldReturnAnEmptyList"?

Reply:
Adding @lcsdtw 's comment, I think it would be interesting to add which method is being tested, like `listAllLocationsShouldReturnAnEmptyList`.

Reply:
Fixed, good catch.